internal/fs/inode/dir.go

```diff
@@ -210,6 +215,10 @@ type dirInode struct {
 	// enabled.
 	prevDirListingTimeStamp time.Time
 	isHNSEnabled            bool
+
+	// Represents if folder has been unlinked in hierarchical bucket. This is not getting used in
+	// non-hierarchical bucket.
+	unlink bool
```

Comment:
unlink is a verb. you need a variable to specify the state. Rename it to unlinked. Look at file.go.